### PR TITLE
Add support for timeout option in pillar file for junos proxy

### PR DESCRIPTION
**Pull request by rsmekala**
### What does this PR do?
Supports timeout option in pillar files, useful for slow devices

### What issues does this PR fix or reference?
https://github.com/Juniper/salt/issues/85

### Tests written?

No

### CODE REVIEW

File: salt/proxy/junos.py [hunk 1]
```diff
@@ -113,6 +113,15 @@ def init(opts):
         thisproxy['initialized'] = False
         return
 
+    if 'timeout' in proxy_keys:
+        timeout = int(opts['proxy']['timeout'])
+        try:
+            thisproxy['conn'].timeout = timeout
+        except Exception as ex:
+            log.error('Not able to set timeout due to: %s' % ex)
```
Comment (s0undt3ch) [inline salt/proxy/junos.py]:
```python
log.error('Not able to set timeout due to: %s', ex)
```

The same for the other log call.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Comment by rsmekala** [comment]
@s0undt3ch I have resolved the lint errors. But CI jobs are still failing due to some unrelated error. Could you please look into it ??

File: salt/proxy/junos.py [hunk 2]
```diff
@@ -113,6 +113,15 @@ def init(opts):
         thisproxy['initialized'] = False
         return
 
+    if 'timeout' in proxy_keys:
+        timeout = int(opts['proxy']['timeout'])
+        try:
+            thisproxy['conn'].timeout = timeout
+        except Exception as ex:
+            log.error('Not able to set timeout due to: {}'.format(ex))
```
Comment (s0undt3ch) [inline salt/proxy/junos.py]: Sorry, no formatting on log calls either.
```
log.error('Not able to set timeout due to: %s', ex)
```

File: salt/proxy/junos.py [hunk 3]
```diff
@@ -113,6 +113,15 @@ def init(opts):
...
+        else:
+            log.debug('RPC timeout set to {} seconds'.format(timeout))
```
Comment (s0undt3ch) [inline salt/proxy/junos.py]: Same as above:
```
log.debug('RPC timeout set to %d seconds', timeout)
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/proxy/junos.py (hunk 3 above)
Comment (rsmekala) [reply salt/proxy/junos.py]: Do you mean `format` function call cannot be used in logging ??

Comment (s0undt3ch) [reply salt/proxy/junos.py]: Yes.
It can, of course, but it shouldn't for performance reasons.

Comment (s0undt3ch) [reply salt/proxy/junos.py]: See the fixed log calls on my commends.

**Review by s0undt3ch — APPROVED** [review]

**Comment by sagetherage** [comment]
already in master

**Pull request merged**